include/storage/flash_map.h

```diff
@@ -75,6 +75,39 @@ struct flash_sector {
 	size_t fs_size; /** flash sector size */
 };
 
+#if defined(CONFIG_FLASH_AREA_CHECK_INTEGRITY)
+/**
+ * @brief Structure for verify flash region integrity
+ *
+ * This is used to pass data and select algorithm (optional) to be used to
+ * check flash integrity.
```

Comment:
The struct does not select or store the algorithm so the doc might be reworked as:

```suggestion
 * This is used to pass data to be used to check flash integrity.
```